x/auth/migrations/legacytx/stdtx.go

```diff
@@ -87,6 +87,7 @@ type StdTx struct {
 	Signatures    []StdSignature `json:"signatures" yaml:"signatures"`
 	Memo          string         `json:"memo" yaml:"memo"`
 	TimeoutHeight uint64         `json:"timeout_height" yaml:"timeout_height"`
+	NonAtomic     bool           `json:"non_atomic" yaml:"non_atomic"`
```

Comment:
If we want to sign a NonAtomic tx with AMINO_JSON, we need to modify the ledger app to show this boolean field in a new screen.

I propose to never allow NonAtomic txs with SIGN_MODE_AMINO_JSON. Does this sound reasonable?

If so, could you add it in this PR, maybe in amino's sign mode handler?

Reply:
Started adding the signing mode validation in https://github.com/osmosis-labs/cosmos-sdk/commit/09eb3a4c4ac70f5e40ad8825e4b51d6fe8845c2b. It's a bit messy because the tx has to be made into a concrete type, which normally doesn't happen in baseapp. 

I'm not sure what a reliable way to test this would be though. Do you know of a good way to generate amino signed txs in tests that I can use for this?